src/rust/ru.desu/src/helper.rs

```diff
@@ -0,0 +1,134 @@
+use crate::constants;
+use aidoku::std::{ValueRef};
+use aidoku::{
+	error::{Result},
+	helpers::uri::encode_uri,
+	prelude::*,
+	std::net::{HttpMethod, Request},
+	Filter, FilterType, Manga, MangaPageResult};
+use alloc::string::{String, ToString};
+use alloc::vec::Vec;
+use crate::parser::debug;
+
+pub fn get_manga_url(id: &str) -> String {
+	format!("{}/{}", constants::BASE_API_URL, encode_uri(id))
+}
+
+pub fn get_chapter_url(manga_id: &str, chapter_id: &str) -> String {
+	format!("{}/chapter/{}", get_manga_url(manga_id), encode_uri(chapter_id))
+}
+
+pub fn fetch_json<T: AsRef<str>>(url: T, fetch_for: &str) -> Result<ValueRef>
+{
+	debug!("Fetching for {}, fetching url {}", fetch_for, url.as_ref());
+
+	let request = Request::new(url, HttpMethod::Get)
+		.header("User-Agent", constants::USER_AGENT)
+		.header("Referer", constants::BASE_URL);
+
+	Ok(request.json()
+		.expect(format!("Failed to find JSON for {}", fetch_for).as_str())
```

Comment:
don't use `expect` (or `unwrap`); if the network fetch fails (e.g. the user isn't connected to the internet), then the source will break until the app is relaunched. you could just use `?` here.